graphics/deko3d/deko_examples/source/SampleFramework/CIntrusiveTree.h

```diff
@@ -129,13 +129,7 @@ class CIntrusiveTreeBase
 template <typename ClassT, typename MemberT>
 constexpr ClassT* parent_obj(MemberT* member, MemberT ClassT::* ptr)
 {
-    union whatever
-    {
-        MemberT ClassT::* ptr;
-        intptr_t offset;
-    };
-    // This is technically UB, but basically every compiler worth using admits it as an extension
-    return (ClassT*)((intptr_t)member - whatever{ptr}.offset);
+    return (ClassT *)((char *)member - (char *)&((ClassT *)0->*ptr));
```

Comment:
This is still UB fwiw. I'd say the original version of this function is easier to read.

Reply:
Reverted.